src/Http/Transport.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * Transport.php
+ *
+ * @since     Jan 2024
+ * @author    Haydar KULEKCI <haydarkulekci@gmail.com>
+ */
+
+namespace Qdrant\Http;
+
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Qdrant\Config;
+
+class Transport implements ClientInterface
+{
+    public function __construct(
+        private readonly ClientInterface $client,
+        private readonly Config $config
+    )
+    {
+    }
+
+    private function prepareHeaders(RequestInterface $request): RequestInterface
+    {
+        $request = $request->withHeader('content-type', 'application/json')
+            ->withHeader('accept', 'application/json')
+            ->withUri(
+                $request->getUri()
+                    ->withHost($this->config->getHost())
+                    ->withPort($this->config->getPort())
+                    ->withScheme('http')
+            );
```

Comment:
`scheme` needs to be solved with a better solution. For testing, I just added here HTTP.

Reply:
Why not add to Config class?

Reply:
This is added as a method into Config.